src/Orleans.Core/Timers/ValueStopwatch.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Diagnostics;
+
+namespace Orleans.Runtime
+{
+    internal struct ValueStopwatch
+    {
+        private static readonly double TimestampToTicks = TimeSpan.TicksPerSecond / (double) Stopwatch.Frequency;
+        private long value;
+
+        public static ValueStopwatch StartNew() => new ValueStopwatch(Stopwatch.GetTimestamp());
+
+        public static ValueStopwatch Zero => default(ValueStopwatch);
+
+        private ValueStopwatch(long timestamp)
+        {
+            this.value = timestamp;
+        }
+
+        public bool IsRunning => this.value > 0;
+        
+        public TimeSpan Elapsed => TimeSpan.FromTicks(this.ElapsedTicks);
+
+        public long ElapsedTicks
+        {
+            get
+            {
+                // A positive timestamp value indicates the start time of a running stopwatch,
```

Comment:
This comment could be document comment, for it to be readable at usage site.

Reply:
Added comments to this and other members, thanks